### PR TITLE
New import manager with converters. New API module StorageApi. Other tiny improvements and bug fixes

### PR DESCRIPTION
**Pull request by almazgimaev**
Main features:
- added new import manager with converters to handle different formats (Supervisely, COCO, YOLO, Cityscapes, PASCAL VOC, from png masks, from CSV (or txt) files, from PDF files), medical 2D, multi-view, and multispectral images: 
    ```python
    converter = sly.ImportManeger(src_dir, sly.ProjectType.IMAGES)
    convert.upload_dataset(dataset_id)
    ``` 
    works with local data, data in Team Files, Cloud Storage
- added new API module: `StorageApi` (single API for all storages: team files, cloud storages, agent folder) + tests

Tiny improvements, features and bug fixes:
- fix in `api.file.upload_directory`: fix paths for cloud storages (missing "/" after `Path.as_posix()`)
- added `progress_cb` in `api.image.upload_multispectral`
- added new `import_settings` field in `ProjectInfo`
- fix in `_set_custom_grouping_settings` in ProjectApi: will update group tag if it does not exist (if exists – will check value type). 
Note: `_set_custom_grouping_settings` is utilized in `api.file.set_multispectral_settings` and `api.file.set_multiview_settings`
- update docker_images/import_export/Dockerfile (added new dependencies for new import)
- new method for volumes: `sly.volume.convert_nifti_to_nrrd(path: str)`

### CODE REVIEW

File: supervisely/convert/converter.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+                total=total,
+                desc=message,
+                unit="B" if is_size else "it",
+                unit_scale=is_size
+            )
+            progress_cb = progress.update
+        return progress, progress_cb
+
+
+# @TODO:
```
Comment (cxnt) [inline supervisely/convert/converter.py:184]: remove?

**Review by NikolaiPetukhov — APPROVED** [review]
LGTM

File: supervisely/api/file_api.py [hunk 2]
```diff
@@ -1337,10 +1337,13 @@ def upload_directory(
...
         local_files = list_files_recursively(local_dir)
-        remote_files = [
-            Path(file.replace(local_dir.rstrip("/"), res_remote_dir.rstrip("/"))).as_posix()
-            for file in local_files
-        ]
+        remote_files = []
+        dir_parts = local_dir.strip("/").split("/")
+        for file in local_files:
+            path_parts = file.strip("/").split("/")
+            remote_file = os.path.join(res_remote_dir, "/".join(path_parts[len(dir_parts) :]))
```
[the remark this answers is not in the document]
Comment (almazgimaev) [reply supervisely/api/file_api.py:1344]:
```python
path_parts = file.strip("/").split("/")
path_parts = path_parts[len(dir_parts) :]
remote_parts = [res_remote_dir.rstrip("/")] + path_parts
remote_file = "/".join(remote_parts)
```

File: supervisely/convert/base_converter.py [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+            new_obj_classes.append(existing_cls)
+        for new_cls in meta2.obj_classes:
+            i = 1
+            new_name = new_cls.name
+            matched = False
+            while meta1.obj_classes.get(new_name) is not None:
+                if meta1.obj_classes.get(new_name).geometry_type == new_cls.geometry_type:
+                    matched = True
+                    break
+                new_name = f"{new_cls.name}_{i}"
```
[the remark this answers is not in the document]
Comment (almazgimaev) [reply supervisely/convert/base_converter.py:263]: done

File: supervisely/convert/base_converter.py [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+    def merge_metas_with_conflicts(
+        self, api: Api, dataset_id: int
+    ) -> Tuple[ProjectMeta, dict, dict]:
+
+        # get meta1 from project and meta2 from converter
+        dataset = api.dataset.get_info_by_id(dataset_id)
+        meta1_json = api.project.get_meta(dataset.project_id)
+        meta1 = ProjectMeta.from_json(meta1_json)
+        meta2 = self._meta or ProjectMeta()
+
```
[the remark this answers is not in the document]
Comment (almazgimaev) [reply supervisely/convert/base_converter.py:251]: done

**Pull request merged**